tests/test_prompt.py

```diff
@@ -103,14 +104,15 @@ def test_create_prompt_appends_and_returns():
     assert "Q-C" in out
 
 
-def test_transform_by_query_basic():
+def test_transform_basic():
     # instruction uses both fields
     instr = lambda query, qcontext: f"ask: {query} | {qcontext}"
     tf = PromptTransformer(instruction=instr, input_fields=["query", "qcontext"])
-    inp = [{"qid": "1", "query": "foo", "qcontext": "bar"}]
-    result = tf.transform_by_query(inp)
+    inp = pd.DataFrame.from_records([{"qid": "1", "query": "foo", "qcontext": "bar"}])
+    result = tf.transform(inp)
```

Comment:
sure, we could have just done tf[{"qid": "1", "query": "foo", "qcontext": "bar"}]), right?

Anyway, leave as is.